linter/conanfile_transform.py

```diff
@@ -0,0 +1,66 @@
+
+# Class ConanFile doesn't declare all the valid members and functions,
+#   some are injected by Conan dynamically to the class.
+
+import textwrap
+import astroid
+from astroid.builder import extract_node, parse
+from astroid.builder import AstroidBuilder
+from astroid.manager import AstroidManager
+
+
+def _settings_transform():
+    module = AstroidBuilder(AstroidManager()).string_build(
+        textwrap.dedent("""
+            class Settings(object):
+                os = None
+                arch = None
+                compiler = None
+                build_type = None
+            """)
+    )
+    return module['Settings']
+
+
+def register(_):
+    pass
+
+def transform_conanfile(node):
+    """Transform definition of ConanFile class so dynamic fields are visible to pylint"""
+
+    str_class = astroid.builtin_lookup("str")
+    dict_class = astroid.builtin_lookup("dict")
+    info_class = astroid.MANAGER.ast_from_module_name("conans.model.info").lookup(
+        "ConanInfo")
+    build_requires_class = astroid.MANAGER.ast_from_module_name(
```

Comment:
```suggestion
    user_info_build_class = astroid.MANAGER.ast_from_module_name("conans.model.user_info").lookup(
        "DepsUserInfo")
    build_requires_class = astroid.MANAGER.ast_from_module_name(
```

Reply:
I don't know why, but using this approach, it fails for `odbc` recipe. The error is weird and I didn't manage to find a fix on the internet. With the `string_build` approach, on the other hand, it works. I'm not very happy about how it is looking like, tbh.